plugins/inventory/falcon_discover.py

```diff
@@ -0,0 +1,378 @@
+            offset = host_lookup["body"]["meta"]["pagination"]["offset"] + max_limit
+            if host_lookup["body"]["meta"]["pagination"]["total"] <= len(host_details):
+                running = False
+
+        return host_details
+
+    def _hostvars(self, host):
+        """Return host variables."""
+        # todo: this is a common list of fields that are returned by the FalconPy API
+        hostvar_opts = [
+            "id",
+            "cid",
+            "aid",
+            "hostname",
+            "entity_type",
+            "first_seen_timestamp",
+            "last_seen_timestamp",
+            "country",
+            "city",
+            "platform_name",
+            "os_version",
+            "kernel_version",
+            "tags",  # list
+            "groups",
+            "agent_version",
+            "external_ip",
+            "current_local_ip",
+            "local_ip_addresses",  # list
+            "reduced_functionality_mode",
+            "mac_address",
+            "fqdn",
+            "location",
+            "state",
+            "confidence",
+            "managed_by",
+            "owned_by",
+            "used_for",
+            "department",
+            "cloud_provider",
+            "cloud_account_id",
+            "cloud_region",
+            "cloud_resource_id",
+            "cloud_registered",
+            "cloud_instance_id",
+        ]
+
+        hostvars = {}
+        for item in hostvar_opts:
+            if item in host:
+                hostvars[item] = host[item]
+
+        return hostvars
+
+    def _get_ip_address(self, hostvars):
+        """Return the IP address for a host."""
+        ip_address = None
+        if "external_ip" in hostvars:
+            ip_address = hostvars["external_ip"]
+        elif "current_local_ip" in hostvars:
+            ip_address = hostvars["current_local_ip"]
+        elif "local_ip_addresses" in hostvars:
+            ip_address = hostvars["local_ip_addresses"][0]
+
+        return ip_address
+
+    def _get_hostname(self, hostvars, ip_address):
+        """Return the hostname for a host."""
+        hostname = hostvars.get("hostname", ip_address)
+
+        return hostname
+
+    def _add_host_to_inventory(self, host_details):
+        """Add host to inventory."""
+        for host in host_details:
+            hostvars = self._hostvars(host)
+            # Only process hosts that have an IP address (reachable)?
+            ip_address = self._get_ip_address(hostvars)
+            if not ip_address:
+                continue
+
+            # Get the hostname
+            hostname = self._get_hostname(hostvars, ip_address)
+
+            # Add the host to the inventory
+            self.inventory.add_host(hostname)
+            self.inventory.set_variable(hostname, "ansible_host", ip_address)
+
+            # Add host variables
+            for key, value in hostvars.items():
+                self.inventory.set_variable(hostname, key, value)
```

Comment:
Use prefix `falcon_discover_` to avoid pollution/potential collision. I guess this would affect your grouping examples?

Reply:
We can probably make this configurable? Like - hostvars_prefix?

Reply:
I would just pick something and use it. Otherwise your doc examples may not match what a user has changed their prefix to, and just make things confusing.